pandas/tests/io/sas/test_sas7bdat.py

```diff
@@ -183,6 +183,18 @@ def test_date_time(datapath):
     tm.assert_frame_equal(df, df0)
 
 
+def test_compact_numerical_values(datapath):
+    # Regression test for #21616
+    fname = datapath("io", "sas", "data", "cars.sas7bdat")
```

Comment:
Using the fixture in this fashion will generate warnings. We just went through an exercise to clean these up in #22515 - can you take a look at that and adjust here accordingly?

Reply:
If I understood the discussion on the pytest-tracker properly, datapath used this way is in fact not something that will be deprecated.

datapath is a "factory as a fixture" as described here:
https://docs.pytest.org/en/latest/fixture.html#factories-as-fixtures

Which is something distinct from what is beeing talked about here:
https://github.com/pytest-dev/pytest/issues/3661

Have I misunderstood?

Reply:
And it produces no warnings, as far as I can tell.

Reply:
This usage is fine I think.

But, do you need to include a binary file in the git repo to reproduce the error on master? I'd like to avoid adding new ones if possible.

Reply:
Hi @TomAugspurger 

I am adding a sas7bdat file for testing the parsing of sas7bdat files. How can that be done more sensibly?